compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -1626,8 +1626,26 @@ impl<'a> Parser<'a> {
     }
 
     /// Parses an optional `move` prefix to a closure-like construct.
-    fn parse_capture_clause(&mut self) -> CaptureBy {
-        if self.eat_keyword(kw::Move) { CaptureBy::Value } else { CaptureBy::Ref }
+    fn parse_capture_clause(&mut self) -> PResult<'a, CaptureBy> {
```

Comment:
You'll need to add a test (e.g. `src/test/ui/parser/incorrect-move-async-order-issue-79694.rs`) with your example (ideally annotated with `// run-rustfix` too).

Reply:
I've added the tests, but I think I missed something.

When running `python ./x.py test src/test/ui`, the test is run, the .stderr is correct and the .fixed is correct.
However, it then can't compile the fixed code.
If I put the fixed code in a normal project, it compiles just fine.
```txt
error: expected one of `!`, `.`, `::`, `;`, `?`, `{`, or an operator, found keyword `move`
  --> C:\Repos\rust\src/test\ui\parser\incorrect-move-async-order-issue-79694.fixed:6:19
   |
LL |     let _ = async move { }; //~ ERROR 6:13: 6:23: the order of `move` and `async` is incorrect
   |                   ^^^^ expected one of 7 possible tokens

error: aborting due to previous error
```
Any idea?

Reply:
Try adding `// edition:2018` to the test.

Reply:
Yes, that worked!